### DISCUSSION TITLE
How to manually import useState from nuxt?

### DISCUSSION THREAD

**Discussion by topd5** [nuxt/nuxt#22268] [Questions] [answered]
Hi.
I try to add tests to my Nuxt 3 project.
I use vitest + @vue/test-utils + happy-dom
Some component uses useState.
Without manual import test gets error:
```
ReferenceError: useState is not defined
```
I tried to add:
```
import { useState } from 'nuxt/app';
```
but got error:
```
TypeError: Package import specifier "#build/app.config.mjs" is not defined in package /home/****/projects//****/node_modules/nuxt/package.json imported from ...
```
How to import useState?

**Comment by joel-wenzel** [comment]
```
<script setup lang="ts">
  import { useState as useState2 } from '#imports'
  useState2()
</script>
```

The alias is just to prove its coming from the manual import and not auto import

**Reply by topd5** [reply]
This solution does not work.
Vitest  and @vue/test-utils know nothing about this alias  '#imports'.

**Reply by joel-wenzel** [reply, marked as answer]
@topd5 Are you using https://github.com/danielroe/nuxt-vitest.
If not give that a shot, i use it and my test using useState works

**Reply by topd5** [reply]
@joel-wenzel , thank you.
Yes, I use regular `vitest`
I'll try to use `nuxt-vitest`
